src/components/message/index.js

```diff
@@ -0,0 +1,89 @@
+import classes from './styles.module.css';
+import { Fragment, useCallback, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import Button from '../button';
+import Input from '../input';
+import SnackBar from '../snackbar';
+import Loader from '../loader';
+import Error from '../error';
+import { isEmptyString } from '../../utils';
+import { SEVERITY } from '../constants';
+import { getMessageUrl } from '../constants';
+
+const Message = props => {
+    const { id } = useParams();
+    const [url, setUrl] = useState(getMessageUrl);
+    const [loading, setLoading] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [errorStatus, setErrorStatus] = useState('');
+    const [decryptMessage, setDecryptMessage] = useState('');
+    const [inputSecretKey, setInputSecretKey] = useState('');
+    const [openSnackbar, setOpenSnackbar] = useState(false);
+    const [snackbarMessage, setSnackbarMessage] = useState('');
+    const [snackbarSeverity, setSnackbarSeverity] = useState(SEVERITY.INFO);
+    const [fetchResult, setFetchResult] = useState(false);
+
+    useEffect(() => {
+        //Check if id exists or not
+        fetch(url + id)
+            .then(response => response.json())
+            .then(result => {
+                setErrorMessage(result.error !== null ? result.error.message : '');
+                setErrorStatus(result.error !== null ? result.status : '');
+                setUrl(result.data !== null ? url + id + '/decrypt' : url);
+            });
+    }, []);
+
+    useEffect(() => fetchResult && fetchMessage(), [fetchResult]);
+
+    const handleInputSecretKeyChange = useCallback(value => setInputSecretKey(value), []);
+    const handleSnackbarClose = useCallback(() => setOpenSnackbar(false), []);
+    const handleRedirectToHome = useCallback(() => props.history.replace('/'), []);
+
+    const handleDecryption = useCallback(() => {
+        const message = isEmptyString(inputSecretKey) && 'Missing Secret Key!';
+        const severity = isEmptyString(inputSecretKey) && SEVERITY.INFO;
+        setSnackbarMessage(message);
+        setSnackbarSeverity(severity);
+        setOpenSnackbar(!isEmptyString(message));
+        setFetchResult(isEmptyString(message) ? true : false);
+    }, [inputSecretKey]);
+
+    const fetchMessage = useCallback(() => {
+        setLoading(true);
+        fetch(url + '?secretKey=' + inputSecretKey)
+            .then(response => response.json())
+            .then(result => {
+                setLoading(false);
+                setErrorMessage(result.error !== null ? result.error.message : '');
```

Comment:
same as above

Reply:
also this is almost similar to above implementation, reduce repetition by taking out fetch part into a different function

Reply:
only 2 lines are same

Reply:
2 lines are same or only 2 of them different?